app/grandchallenge/core/context_processors.py

```diff
@@ -31,6 +32,11 @@ def challenge(request):
         "challenge_perms": get_perms(user, challenge),
         "user_is_participant": challenge.is_participant(user),
         "pages": challenge.page_set.all(),
+        "num_admins": challenge.get_admins().count(),
+        "num_participants": challenge.get_participants().count(),
+        "num_requests": challenge.registrationrequest_set.filter(
+            status=RegistrationRequest.PENDING
+        ).count(),
```

Comment:
These are really expensive queries that as it's in a context processor will be performed on every request. To me, this information it is not worth 3 extra db queries on every challenge page. If needed, this information should be either:

- stored in fields on the challenge objects (big change, needs tracking)
- only evaluated in the templates so that only admins get the 3 extra queries (slow, but doesn't effect everyone else)

Reply:
Ok, makes sense. I can move these queries to the challenge,html template so that they only get evaluated for admins, but it's not necessary to have these badges at all, so we can totally leave them out. The only one that is maybe a useful addition from the admin perspective is the count of open requests, then admins only need to open the request approval page if there are pending approvals, but this is only relevant for challenges where participants are not automatically accepted, so again probably just a tiny gain in terms of user friendliness.